clib/cArray.mli

```diff
@@ -80,12 +80,25 @@ sig
   (** [chop i a] returns [(a1, a2)] s.t. [a = a1 + a2] and [length a1 = n].
       Raise [Failure "Array.chop"] if [i] is not a valid index. *)
 
-  val smartmap : ('a -> 'a) -> 'a array -> 'a array
-  (** [smartmap f a] behaves as [map f a] but returns [a] instead of a copy when
+  val smart_map : ('a -> 'a) -> 'a array -> 'a array
+  (** [smart_map f a] behaves as [map f a] but returns [a] instead of a copy when
       [f x == x] for all [x] in [a]. *)
 
+  val smart_map2 : ('a -> 'b -> 'b) -> 'a array -> 'b array -> 'b array
+  (** [smart_map2 f a b] behaves as [map2 f a b] but returns [a] instead of a copy when
+      [f x y == y] for all [x] in [a] and [y] in [b] pointwise. *)
+
+  val smart_fold_left_map : ('a -> 'b -> 'a * 'b) -> 'a -> 'b array -> 'a * 'b array
+  (** Same as [smart_map] but threads an additional state left-to-right. *)
+
+  val smart_fold_left2_map : ('a -> 'b -> 'c -> 'a * 'c) -> 'a -> 'b array -> 'c array -> 'a * 'c array
+  (** Same as [smart_map2] but threads an additional state left-to-right. *)
+
+  val smartmap : ('a -> 'a) -> 'a array -> 'a array
+  (** @deprecated Same as [smart_map] *)
+
   val smartfoldmap : ('r -> 'a -> 'r * 'a) -> 'r -> 'a array -> 'r * 'a array
-  (** Same as [smartmap] but threads an additional state left-to-right. *)
+  (** @deprecated Same as [smart_fold_left_map] *)
```

Comment:
This should be `[@@ocaml.deprecated "Please use [smart_fold_left_map]"]` Ditto for the rest of definitions.

Reply:
Done.